components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -204,7 +210,8 @@ void SolanaTxManager::OnGetSignatureStatuses(
       continue;
 
     if (!signature_statuses[i]) {
-      if (meta->tx()->message()->last_valid_block_height() < block_height) {
+      if (meta->tx()->message()->last_valid_block_height() &&
```

Comment:
For transactions created from dapps, we can't get this info so last_valid_block_height will be kept as 0 (the initialized value).
Created a follow-up issue to support dapp transactions by calling IsBlockhashValid at https://github.com/brave/brave-browser/issues/23120.